manifests/jail.pp

```diff
@@ -52,6 +59,13 @@
   }
 
   # Jail creation
+  file { "${fail2ban::params::config_dir_path}/jail.d":
+    ensure => directory,
+    owner  => $config_file_owner,
+    group  => $config_file_group,
+    before => File["custom_jail_${name}"]
+  }
+
   file { "custom_jail_${name}":
     ensure  => file,
     path    => "${fail2ban::params::config_dir_path}/jail.d/${name}.conf",
```

Comment:
That reference to params is not right, pass it in from init like for instance `$config_file_owner`

Reply:
Sorry - that's not your code, the module is already wrong here :-)